pyrene/src/components/Filter/FilterPopover/FilterPopover.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { FunctionComponent, MouseEvent } from 'react';
+
+import styles from './filterPopover.css';
+import ButtonBar from '../../ButtonBar/ButtonBar';
+import Button from '../../Button/Button';
+import FilterOption from '../FilterOption/FilterOption';
+import Collapsible from '../../Collapsible/Collapsible';
+import { Filter, FilterValues, HandleFilterChange } from '../types';
+
+export interface FilterPopoverProps {
+  filterNegatedKeys: string[],
+  filters: Array<Filter>,
+  filterValues: FilterValues,
+  handleFilterChange: HandleFilterChange,
+  negatable: boolean,
+  onClose: (e: MouseEvent) => void,
+  onFilterApply: (e: MouseEvent) => void,
+  onFilterClear: (e: MouseEvent) => void,
+}
+
+const FilterPopover: FunctionComponent<FilterPopoverProps> = ({
+  handleFilterChange,
+  filterValues,
+  negatable,
+  filters,
+  filterNegatedKeys,
+  onFilterClear,
+  onFilterApply,
+  onClose = () => null,
+}: FilterPopoverProps) => (
+  <div className={styles.filterPopover}>
+    <div className={styles.title}>Select Filter</div>
+    {negatable && <div className={styles.negateTitle}>Negate</div>}
+    <div className={styles.filterOptions}>
+      {filters.length <= 6
+        ? filters.map((filter) => (
+          <FilterOption {...filter}
+            value={filterValues ? filterValues[filter.id] : null}
```

Comment:
Sure null and not undefined?
Null does not allow for/overwrite a default value.

Reply:
> Sure null and not undefined?
> Null does not allow for/overwrite a default value.

@justmejulian Thanks for taking some time for that. In this PR I dit not touch on the logic. So it s just about TypeScript conversion. See the previous file.

<img width="867" alt="Screenshot 2021-05-21 at 09 23 34" src="https://user-images.githubusercontent.com/6510794/119098726-6ca91800-ba16-11eb-99e9-0742835fd0a6.png">

Reply:
Sure but if we find bugs then why not fix them 😊
Not saying it is a bug just saying it might be worth looking at if we are already touching this component.

Also this is probably the reason why [ value?: InputValue | null,](https://github.com/open-ch/pyrene/pull/115/files#diff-e1681eef36baa2d2f1c8ec8af4c7aa3ba704be3768375af68053753ff1e87423L24) was set like that. Why also null and not just optional?
My guess is to allow exactly this usage.

The thing is that proptypes did not complain if you passed null to an optional prop, however typescript does because we have `strictNullChecks` enabled. So why do we hack null as an option to avoid errors that are shown because we use `strictNullChecks`?

Reply:
> The thing is that proptypes did not complain if you passed null to an optional prop, however typescript does because we have `strictNullChecks` enabled. So why do we hack null as an option to avoid errors that are shown because we use `strictNullChecks`?

@justmejulian Make sense. Thanks for your valuable remark. Let me check.